test/parallel_api/iterator/permutation_iterator.h

```diff
@@ -207,10 +185,20 @@ struct test_through_permutation_iterator<TSourceIterator, TSourceDataSize, perm_
     {
         using ValueType = typename ::std::iterator_traits<TSourceIterator>::value_type;
 
-        auto no_transformation = [](ValueType val) { return val; };
+        // Attention: we are unable to use lambda here due compile errors in tbb backend: lambda isn't default constructible.
```

Comment:
This comment looks like blaming TBB backend for compilation issues. As we discussed, the default constructability requirement is part of the C++ standard and is not an issue of the TBB backend. So I would better rephrase to something like:
```suggestion
        // Using callable object instead of lambda here to ensure transform iterator would be
        // default constructible, that is part of the Forward Iterator requirements in the C++ standard.
```

Reply:
Applied.